src/App.tsx

```diff
@@ -1,61 +1,103 @@
+import React, { useState } from 'react';
 import './App.scss';
+import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import { TodoList } from './components/TodoList/TodoList';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+export const App: React.FC = () => {
+  const [todos, setTodos] = useState(todosFromServer);
+  const [title, setTitle] = useState('');
+  const [userId, setUserId] = useState(0);
+  const [errors, setErrors] = useState({ title: '', userId: '' });
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    const newErrors = { title: '', userId: '' };
+    let hasError = false;
+
+    if (!title.trim()) {
+      newErrors.title = 'Please enter a title';
+      hasError = true;
+    }
+
+    if (!userId) {
+      newErrors.userId = 'Please choose a user';
+      hasError = true;
+    }
+
+    if (hasError) {
+      setErrors(newErrors);
+      return;
+    }
+
+    const newTodo = {
+      id: todos.length ? Math.max(...todos.map(todo => todo.id)) + 1 : 1,
+      title: title.trim(),
+      userId,
+      completed: false,
+    };
+
+    setTodos([...todos, newTodo]);
+    setTitle('');
+    setUserId(0);
+    setErrors({ title: '', userId: '' });
+  };
+
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const value = event.target.value.replace(/[^a-zA-Z0-9\s]/g, '');
+    setTitle(value);
+    if (errors.title) setErrors(prev => ({ ...prev, title: '' }));
+  };
+
+  const handleUserChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setUserId(Number(event.target.value));
+    if (errors.userId) setErrors(prev => ({ ...prev, userId: '' }));
+  };
 
-export const App = () => {
   return (
     <div className="App">
-      <h1>Add todo form</h1>
+      <h1>Add TODO Form</h1>
 
-      <form action="/api/todos" method="POST">
+      <form onSubmit={handleSubmit}>
         <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
+          <label htmlFor="titleInput">Title</label>
+          <input
+            type="text"
+            id="titleInput"
+            data-cy="titleInput"
+            placeholder="Enter a TODO title"
+            value={title}
+            onChange={handleTitleChange}
+          />
+          {errors.title && <span className="error">{errors.title}</span>}
         </div>
 
         <div className="field">
-          <select data-cy="userSelect">
+          <label htmlFor="userSelect">User</label>
+          <select
+            id="userSelect"
+            data-cy="userSelect"
+            value={userId}
```

Comment:
The default value for `userId` is `0`. Ensure that this aligns with the user data from `usersFromServer`, as user IDs typically start from `1`. If `0` is not a valid user ID, consider using a different default value.